Mathlib/Topology/KrullDimension.lean

```diff
@@ -0,0 +1,23 @@
+/-
+Copyright (c) 2024 Jujian Zhang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jujian Zhang, Fangming Li
+-/
+import Mathlib.Order.KrullDimension
+import Mathlib.Topology.Irreducible
+
+/-!
+# The Krull dimension of a topological space
+
+The Krull dimension of a topological space is the order theoretic Krull dimension applied to the
+collection of all its subsets that are closed and irreducible. Unfolding this definition, it is
+the length of longest series of closed irreducible subsets ordered by inclusion.
```

Comment:
Why not add some TODOs? For example Krull dimension of Spec(R) is Krull dimension of R, for R a commutative ring (do we have Krull dimension of a commutative ring?)

Reply:
I added the TODO comment. That result is proven in #6277, which is too big, but @FMLJohn  is working on splitting it into smaller PRs.